### PR TITLE
pom-experiments

### PR DESCRIPTION
**Pull request by timk01**

### CODE REVIEW

File: telegram-bot/pom.xml [hunk 1]
```diff
@@ -40,12 +40,13 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>31.1-jre</version>
+            <version>${guava.version}</version>
```
Comment (LostKali) [inline telegram-bot/pom.xml:43]: what is the need for guava?

File: buyer/pom.xml [hunk 2]
```diff
@@ -23,6 +23,59 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
```
Comment (LostKali) [inline buyer/pom.xml:27]: if dependencies exists in the parent pom there is no need to declare them

File: buyer/src/main/java/ru/otus/buyer/service/BuyerService.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class BuyerService {
+
+    private final BuyerRepository buyerRepository;
+    private final Converter<BuyerRecord, Buyer> converter;
+
+    public void saveUserInfo(BuyerRecord buyerRecord) {
+        buyerRepository.save(Objects.requireNonNull(converter.convert(buyerRecord)));
```
Comment (LostKali) [inline buyer/src/main/java/ru/otus/buyer/service/BuyerService.java:20]: is there any possibility converter produces null?

File: buyer/src/main/resources/application.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+spring:
+  datasource:
+    url: jdbc:h2:file:C:/data/buyer
```
Comment (LostKali) [inline buyer/src/main/resources/application.yml:3]: remove C:/ - it will create a storage in the data/buyer foler

File: pom.xml [hunk 5]
```diff
@@ -20,6 +20,17 @@
 
     <properties>
         <java.version>17</java.version>
+        <starter.data.jpa.version>2.7.5</starter.data.jpa.version>
```
Comment (LostKali) [inline pom.xml:23]: let's keep it like this, added tech task for the future `[tech] maven dependency and dependencyManagement`

File: telegram-bot/pom.xml (hunk 1 above)
Comment (timk01) [reply telegram-bot/pom.xml:43]: No idea atm. Presumably, it was used in tests or as additional lib, but i don't remember why and when.
Googled and still no idea. Deleted, tests and project work.

File: buyer/src/main/java/ru/otus/buyer/service/BuyerService.java (hunk 3 above)
Comment (timk01) [reply buyer/src/main/java/ru/otus/buyer/service/BuyerService.java:20]: изображение

Comment (timk01) [reply buyer/src/main/java/ru/otus/buyer/service/BuyerService.java:20]: So, rather omit any check.

Comment (timk01) [reply buyer/src/main/java/ru/otus/buyer/service/BuyerService.java:20]: Or is it linken with question "why record" ?

**Review by LostKali — APPROVED** [review]

**Pull request merged**